TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml.cs

```diff
@@ -53,9 +100,54 @@ public async Task<IActionResult> OnGet()
                 TimelineItemType.Task,
                 t.ModifiedOn.WithDqtBstFix(isLocalTime: true)!.Value,
                 t)))
+            .Concat(eventsWithUser.Select(MapTimelineEvent))
             .OrderByDescending(i => i.Timestamp)
             .ToArray();
 
         return Page();
     }
+
+    public TimelineItem MapTimelineEvent(EventWithUser eventWithUser)
+    {
+        var assembly = typeof(EventBase).Assembly;
+        var eventTypeName = typeof(EventBase).Namespace + "." + eventWithUser.EventName;
+        var eventType = assembly.GetType(eventTypeName);
+        var @event = JsonSerializer.Deserialize(eventWithUser.EventPayload, eventType!);
```

Comment:
I think should move this into a method on `EventBase` with a corresponding `Serialize` method to centralise it. Can then call it from `Event.FromEventBase` and `Event.ToEventBase` too. Using the `JsonSerializerOptions` on `EventBase` will sort out `RaisedBy` correctly too.
```cs
public static EventBase Deserialize(string payload, string eventName) ...

public string Serialize()...
```

Reply:
Good idea .. doing that now

Reply:
Done